### ISSUE TITLE
require rest tracking token in x-goog-api-client

### ISSUE THREAD

**Issue by vchudnov-g** [googleapis/gapic-showcase#1176]
Right now we are only warning in server logs when the `rest/XXX` token is not present in REST requests:
```
REST request failed format check: (HeaderAPIClientError) did not find expected HTTP header "X-Goog-Api-Client": "rest/"
```

We need this for tracking, so we should make this a hard requirement and fail the HTTP request if the token is not present.

**Comment by vchudnov-g** [comment]
@noahdietz FYI. LMK of any concerns.

(sorry for tagging you erroneously on the other repo)

**Comment by noahdietz** [comment]
> > > our main REST code is in the library itself, so it probably makes the most sense to use the gapic-generator-python version number there
> > 
> > The version of the generator that generated that client? Or the version of the client library itself?
>
> ...yes, version of the generator

Where the "majority" of the REST logic resides is language dependent. Go puts all of it in the GAPIC, whereas Java puts it mostly in gax-java. So the version of the generator the was used to generate the client executing the request won't be a definitive answer to "which version of our REST logic executed this request?". 

As such, I think a different value should be used for this `rest/` key or it should be dropped.

**Comment by vchudnov-g** [comment]
Sorry, to clarify: we would not be checking the version number `XXX` itself, because that is indeed language-dependent. (The answer you quoted was specific to Python.) But we would be requiring that the `rest/XXX` token be present with any value of `XXX`. We need this to measure which clients/transports people are using for various services. 

We discussed and approved the `rest/` token earlier; the only change I'm specifying here is making Showcase require it rather than simply warn in logs like it does currently.

**Comment by noahdietz** [comment]
In that case, sure let's do it.

**Comment by vchudnov-g** [comment]
Well, nm! This was already implemented as a requirement in #714!
